src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -677,3 +701,70 @@ def get_entry_to_persist(self, sp_id, tenant):
             entry[_SERVICE_PRINCIPAL_CERT_THUMBPRINT] = self.thumbprint
 
         return entry
+
+
+class MsiCreds(object):
+    MSI_ACCOUNT_NAME = 'MSI'
+
+    @staticmethod
+    def split_user_info(user):
+        import uuid
+        parts = user.split('@')
+        if len(parts) == 2:
+            try:
+                uuid.UUID(parts[0])  # valid subscription id?
+                int(parts[1])  # valid port?
+                return parts[1], parts[0]
+            except ValueError:
+                pass
+        return None, None
+
+    @staticmethod
+    def get_tenant_id(arm_endpoint, subscription_id):
+        import re
+        import requests
+        url = '{}/subscriptions/{}?api-version=2014-04-01'.format(arm_endpoint.rstrip('/'), subscription_id)
+        logger.debug('MSI: Retrieving tenant id by invoking GET from %s', url)
+        result = requests.get(url)
+        if result.status_code != 401:
+            raise CLIError('Failed to retrieve the tenant id of subscription {}'.format(subscription_id))
+        exp = r'.+authorization_uri="(.*?)".+'
+        r = re.match(exp, result.headers['www-authenticate'])
+        tenant_id = r.group(1).split('/')[-1]
+        logger.debug('MSI: Got tenant id: %s', tenant_id)
+        return tenant_id
+
+    @staticmethod
```

Comment:
Is there a reason these need to be static methods?  Couldn't you just init the Creds object with parameters and call get token on the Creds object?  Otherwise I don't see why this is a class.